extensions/wikia/UserLogin/UserSignupSpecialController.class.php

```diff
@@ -621,36 +634,52 @@ public function formValidation() {
 				break;
 		}
 
-		$this->result = ( $signupForm->msgType == 'error' ) ? $signupForm->msgType : 'ok' ;
+		$this->result = ( $signupForm->msgType == 'error' ) ? $signupForm->msgType : 'ok';
 		$this->msg = $signupForm->msg;
 		$this->errParam = $signupForm->errParam;
 	}
 
 	/**
 	 * Disables User Signup Captcha for automated tests, mobile skin, and sites such as internal
-	 *
-	 * @throws MWException
 	 */
 	private function disableCaptcha() {
+		// Disable captcha for automated tests and wikia mobile and sites disabling it, e.g. Internal
+		if ( $this->shouldDisplayCaptcha() ) {
```

Comment:
This function was already named poorly given that it was called ```disableCaptcha``` but then had several tests for when to not disable captcha.  It definitely violates the principle of least surprise.

I agree that the check should move to the constructor and I think it should also be renamed to what its actually doing; checking if we should *disable* captcha (given that it starts with the assumption that captcha is enabled and then tests two cases that would break that assumption.)  That way we wouldn't have to test a negative and the constructor would read more clearly:
```php
if ( $this->shouldDisableCaptcha ) {
    $this->disableCaptcha();
}
```
Also, I think when this is refactored ```disableCaptcha``` should probably move out of the ```UserSignupSpecialController``` and into ```CaptchaController```.